File: semester2/5_Fuzzy-Software-Technologies/aics-fuzzy-software-technologies/ticket-backend/src/main/java/aics/domain/movie/dtos/MovieListItemDto.java
package aics.domain.movie.dtos;

import aics.domain.movie.entities.Movie;
import lombok.Data;
import lombok.experimental.Accessors;

import java.io.Serializable;
import java.util.Base64;

@Data
@Accessors(chain = true)
public class MovieListItemDto implements Serializable {
    private Long movieId;
    private String name;
    private String description;
    private String image;
    private String imageName;
    private String imageMimePrefix;
    private int year;
    private double rating;
    private int popularity;
    private int duration;
    private double topsisScore;

    public static MovieListItemDto fromMovie(Movie movie) {
        if (movie == null) {
            return null;
        }
        return new MovieListItemDto()
                .setMovieId(movie.getMovieId())
                .setName(movie.getName())
                .setDescription(movie.getDescription())
                .setImage(Base64.getEncoder().encodeToString(movie.getImage()))
                .setImageName(movie.getImageName())
                .setImageMimePrefix(movie.getImageMimePrefix())
                .setYear(movie.getYear())
                .setRating(movie.getRating())
                .setPopularity(movie.getPopularity())
                .setDuration(movie.getDuration())
                .setTopsisScore(0);
    }


    public static MovieListItemDto fromMovieAndTopsisScore(Movie movie, double topsisScore) {
        if (movie == null) {
            return null;
        }
        return new MovieListItemDto()
                .setMovieId(movie.getMovieId())
                .setName(movie.getName())
                .setDescription(movie.getDescription())
                .setImage(Base64.getEncoder().encodeToString(movie.getImage()))
                .setImageName(movie.getImageName())
                .setImageMimePrefix(movie.getImageMimePrefix())
                .setYear(movie.getYear())
                .setRating(movie.getRating())
                .setPopularity(movie.getPopularity())
                .setDuration(movie.getDuration())
                .setTopsisScore(topsisScore);
    }
}

File: semester2/5_Fuzzy-Software-Technologies/aics-fuzzy-software-technologies/ticket-frontend/src/modules/movie/dtos/movie-list-item-dto.ts
import { TypeUtils } from '../../core/type-utils';

export class MovieListItemDto {
    movieId: number = 0;
    name: string = '';
    description: string | null = null;
    image: string = '';
    imageName: string = '';
    imageMimePrefix: string = '';
    year: number = 0;
    rating: number = 0;
    popularity: number = 0;
    duration: number = 0;
    topsisScore: number = 0;

    static fromObj(obj: any): MovieListItemDto | null {
        if (!obj) {
            return null;
        }
        const movieModel: MovieListItemDto = new MovieListItemDto();
        movieModel.movieId = obj.movieId;
        movieModel.name = obj.name;
        movieModel.description = obj.description;
        movieModel.image = obj.image;
        movieModel.imageName = obj.imageName;
        movieModel.imageMimePrefix = obj.imageMimePrefix;
        movieModel.year = obj.year;
        movieModel.rating = obj.rating;
        movieModel.popularity = obj.popularity;
        movieModel.duration = obj.duration;
        movieModel.topsisScore = obj.topsisScore;
        return movieModel;
    }

    static listFromObjList(objs: any[]): MovieListItemDto[] {
        if (!objs) {
            return [];
        }
        return objs.map(MovieListItemDto.fromObj).filter(TypeUtils.isNonNullable);
    }
}